core/src/main/java/org/apache/cxf/staxutils/transform/InTransformReader.java

```diff
@@ -261,13 +261,22 @@ private void handleDefaultMode(QName name, QName expected) {
     }
 
     private void handleDeepDrop() throws XMLStreamException {
-        final int depth = getDepth();
-        while (depth != getDepth() || super.next() != XMLStreamConstants.END_ELEMENT) {
-            // get to the matching end element event
+        int read = 0;
+        while (hasNext()) {
+            // get to the matching end element event (accounting for all inner elements)
+            final int event = super.next();
+            if (event == XMLStreamConstants.START_ELEMENT) {
+                ++read;
+            } else if (event == XMLStreamConstants.END_ELEMENT) {
+                if (read == 0) {
```

Comment:
since break never executes else:

```
                if (read == 0) {
                    break; 
                }
                read--;
```

Reply:
Matter of preferences, I will keep it the way it is now